assembly/src/procedures/mod.rs

```diff
@@ -162,12 +162,14 @@ impl AsRef<str> for ProcedureName {
 impl Serializable for ProcedureName {
     fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
         let name_bytes = self.name.as_bytes();
-        let num_bytes = name_bytes.len() as u8;
-        if num_bytes > MAX_LABEL_LEN {
```

Comment:
here `num_bytes` is `usize` and it was cast to `u8`, this is the comparison I mentioned in the PR description.